Single post options

cristian-ungureanu opened the pull request:
### Summary
- Added the single post options according to the [design](https://www.figma.com/file/LBW82Rm2SA5rvvLdKrRnBl/Neve-Core-UI?node-id=296%3A7887). Some controls were changed during the implementation based on what I discussed with John and Marius
- Besides the new options for the single post, the controls for Blog meta that are in Pro, were moved in lite.

### Will affect visual aspect of the product
NO

### Test instructions
- There are some controls that are added in Pro. Check https://github.com/Codeinwp/neve-pro-addon/pull/1374
- First of all, you need to change `NEVE_VERSION` variable to be `3.0.0`. There is a check in Neve Pro that verify that variable and if it's lower than 3.0, the blog meta will be added from the pro, else they will be added from lite 
- Check that all the options are working as they should be
- Check the meta options for Blog too.
- The meta options from the single post have their default values from blog options. If you don't change the single ones, the ones from the blog should apply.

@abaicus I've created a new CSS file as you've asked. It's named `single-new.scss`.
I also made some small modifications in `_blogpost-meta.scss` and in `_comments.scss`.
I modified the `_media-queries.scss` too so I can hook responsive classes.

Closes Codeinwp/neve-pro-addon#1110.

selul commented on globals/sanitize-functions.php:
```diff
@@ -206,3 +206,132 @@ function neve_sanitize_typography_control( $value ) {
+ *
+ * @return array
+ */
+function sanitize_alignment( $input ) {
```
please either use prefixed functions or namespace, sanitize_alignment is too general

cristian-ungureanu replied:
I just moved it from here https://github.com/Codeinwp/neve/blob/4186d70efed6b96f9f7476d8451fba6a1c28eb71/header-footer-grid/Core/Components/Abstract_Component.php but now that is global it needs a prefix, you're right. Thanks for the feedback!

irinelenache commented:
@cristian-ungureanu Nice work 🚀 Tested it and found some small things:

- [ ] The boxed layout **background color** applies only on customizer
customizer: https://prnt.sc/13muc37
frontend: https://prnt.sc/13mue8i

- [ ] The spacing between elements isn't applied between **Comments** and **Send comment** (i don't know if this is the desired behaviour) https://prnt.sc/13mvc3z

Will come back after more testing 🔜

cristian-ungureanu commented:
Thank you @irinelenache, I updated the defaults, they should work ok now. About the second issue, it's the desired behavior. If the users want more space between comments and forms, they can enable the boxed layout and control the padding there.

cristian-ungureanu commented:
@irinelenache I've made the changes I was talking with you about on Slack.

There are 3 scenarios that you should test:
- Lite production + Pro new version
- Lite new version + Pro production
- Lite new version + Pro new version

You will have all the new controls only in the last scene. 

1. In the first scenario, only the controls that are new and added to the Pro version will be available.
 - Comments: just the title and comments style ( always show or show/hide mechanism ) will be available
 - Sharing: all the new controls from the design should be there
 - Author box: all the new controls from the design will be there except the boxed controls. They are added in lite and used from there
 - Related posts:  all the new controls from the design will be there except the boxed controls.

2. In the second scenario, only the controls that are new and added to the Lite version and the ones that are already in the Pro version will be available
- The new controls for cover/title layout
- All the controls from the design for elements ordering
- All the controls from the design for post meta 
- All the controls from the design for comments, except the show/hide mechanism that it's already added in pro.

You told me on Slack that there is still an issue with the defaults but I couldn't find any. Please make a fresh instance and test it again. If the issue persists, a video or access to an instance where you manage to replicate the problem would be great.

Don't forget to change the NEVE_VERSION variable in functions.php to be 3.0.0 when you test in the third scenario.

When you test in the third scenario, please make sure the meta controls on the blog still work.
Also, please make sure that the title, meta, and thumbnail elements from single post options ( the one from Neve sidebar when you edit the post ) disappear when the cover layout is set.

irinelenache commented:
@cristian-ungureanu Thanks for the instructions 🙏 
About the problem i reported, it still happens for me, i tried to create a fresh instance and downloaded the latest builds again. You can check here: 
```
https://reduced-aphid.jurassic.ninja
demo
oMOPSIE2zLnZ
```

cristian-ungureanu commented:
Thanks, @irinelenache, I just fixed it.

irinelenache commented:
@cristian-ungureanu now the problem is fixed 👍 

- [ ] When i set the cover layout, the title, meta, and thumbnail elements from single post options are still are still visible in Neve options. They are deactivated and activating them doesn't make any change (i suppose they shouldn't be visible at all)
editor: https://prnt.sc/13uo0ce
frontend: https://prnt.sc/13uo2zi

cristian-ungureanu commented:
Thank you @irinelenache, it should work ok now. I just had to rebuild the meta sidebar part.

irinelenache commented:
@cristian-ungureanu I don't know if this is expected: The **spacing between elements control** is positioned at the half (https://prnt.sc/13zhqc1), but when i move it a bit, the spacing becomes very big (https://prnt.sc/13zhvjb). When i click on **Reset**, the default value is 20 and the control is positioned here (https://prnt.sc/13zinud). Should the control be positioned where it is now when the value isn't changed?

cristian-ungureanu commented:
@irinelenache about the spacing control, that's the normal behavior. The control can't have a default and its range is between 0 and 500px. The range slider is made to stay in the middle at the beginning so as soon as you drag it, it will go from its default value from CSS to 250+px. Not much to do, except to make the limit a bit smaller but I think it's ok as it is now.

The other two issues you've mentioned happened because you didn't change the NEVE_VERSION constant in function.php to 3.0.0.

irinelenache commented:
@cristian-ungureanu now all scenarios work fine on my end 🚀

cristian-ungureanu commented:
Thank you @irinelenache for testing this.

@abaicus let me know if you need any help with the scss rewriting.
